src/common/types.cpp

```diff
@@ -943,12 +1200,30 @@ LogicalType::~LogicalType() {
 void LogicalType::Serialize(Serializer &serializer) const {
 	serializer.Write<LogicalTypeId>(id_);
 	ExtraTypeInfo::Serialize(type_info_.get(), serializer);
+	if (id_ == LogicalTypeId::ENUM) {
+		auto size = EnumType::GetSize(*this);
```

Comment:
This special case here shouldn't be necessary, no? This can all be handled in the ExtraTypeInfo.